### ISSUE TITLE
Pyright crashes in VSCode when opening a git diff view

### ISSUE THREAD

**Issue by mtreca** [microsoft/pyright#6787]
**Describe the bug**
Opening any Python file in the source control view in VSCodium crashes Pyright with the following traceback:

```
[Info  - 12:20:09 PM] Pyright language server 1.1.342 starting
[Info  - 12:20:09 PM] Server root directory: file:///Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist
[Info  - 12:20:09 PM] Starting service instance "REPO_NAME"
Received pythonPath from Python extension: /Users/maximet/.venv-py38/REPO_NAME/bin/python
[Info  - 12:20:09 PM] Setting pythonPath for service "REPO_NAME": "/Users/maximet/.venv-py38/REPO_NAME/bin/python"
[Info  - 12:20:09 PM] Loading pyproject.toml file at /Users/maximet/Code/REPO_NAME/pyproject.toml
[Info  - 12:20:09 PM] Pyproject file "/Users/maximet/Code/REPO_NAME/pyproject.toml" has no "[tool.pyright]" section.
[Info  - 12:20:09 PM] Assuming Python version 3.8
[Info  - 12:20:15 PM] Found 11004 source files
Error: Debug Failure. git:/Users/maximet/Code/REPO_NAME/FILE.py?%7B%22path%22%3A%22%2FUsers%2Fmaximet%2FCode%2REPO_NAME%2FFILE.py%22%2C%22ref%22%3A%22~%22%7D is not a file based URI.
    at d.getFilePath (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/common/uri/webUri.ts:164:15)
    at S.isInZip (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/common/realFileSystem.ts:415:26)
    at l.isInZip (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/readonlyAugmentedFileSystem.ts:159:28)
    at canNavigateToFile (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/languageService/navigationUtils.ts:14:16)
    at b.canNavigateToFile (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/languageServerBase.ts:1410:16)
    at forEach (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/languageServerBase.ts:1304:23)
    at Array.forEach (<anonymous>)
    at b.onAnalysisCompletedHandler (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/languageServerBase.ts:1303:29)
    at c._onAnalysisCompletion (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/languageServerBase.ts:441:57)
    at c.handleAnalysisResponse (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/backgroundAnalysisBase.ts:245:22)
    at MessagePort.<anonymous> (/Users/maximet/.vscode-oss/extensions/ms-pyright.pyright-1.1.342/dist/pyright-internal/src/backgroundAnalysisBase.ts:279:61)
    at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:735:20)
    at MessagePort.exports.emitMessage (node:internal/per_context/messageport:23:28)
```

It seems to me that Pyright struggles with identifying the git diff as a file on disk (since it doesn't exist).

**Code or Screenshots**

This should be reproducible with any python file in VSCodium and opening the source control view. I have tried it with a simple script in a git repo:

```py
from __future__ import annotations

import os
from dotenv import find_dotenv, load_dotenv

def load_secrets():
    if dotenv_path := find_dotenv():
        load_dotenv(dotenv_path=dotenv_path)


def get_secret(key: str) -> str | None:
    return os.getenv(key=key)


load_secrets()
```

**VS Code extension or command-line**
I am running Pyright latest in VSCodium. No Pylance used.

**Comment by mtreca** [comment]
I am using 1.1.342 indeed.

**Comment by erictraut** [comment]
@rchiodo, this looks like another regression related to URIs. Would you like to take it?

**Comment by debonte** [comment]
Rich is OOF. I'll take a look at this.

**Comment by debonte** [comment]
Other scenarios that are impacted:
- Opening a notebook (#6789)
- Opening a file from a GitHub repository via the `Remote Repositories: Open Remote Repository..." command from the "GitHub Repositories" extension

Despite this issue and #6789 getting opened on the same day, this is not a recent regression. These issues have been present since the big URI refactoring in 1.1.340.

**Comment by mtreca** [comment]
Any idea if this could also affect code actions on save? Pyright hasn’t been playing nice with other extensions with that regard recently as well. I can open another issue later as well if unrelated.

**Comment by debonte** [comment]
> Pyright hasn’t been playing nice with other extensions with that regard recently as well. I can open another issue later as well if unrelated.

Are you seeing a crash in `getFilePath` in the Pyright log? If not, it's a different issue.

**Comment by debonte** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.343, which I just published. It will also be included in a future release of pylance.

**Comment by mtreca** [comment]
New version fixes it indeed. Thanks a ton for the quick fix!
